tla/ccfraft.tla

```diff
@@ -121,98 +132,248 @@ VARIABLE configurations
 \* TODO: re-add a server in a raft_scenario test?
 VARIABLE removedFromConfiguration
 
-reconfigurationVars == <<reconfigurationCount, removedFromConfiguration, configurations>>
+RemovedFromConfigurationTypeInv ==
+    removedFromConfiguration \subseteq Servers
+
+reconfigurationVars == <<
+    reconfigurationCount, 
+    removedFromConfiguration, 
+    configurations
+>>
+
+ReconfigurationVarsTypeInv ==
+    /\ ReconfigurationCountTypeInv
+    /\ ConfigurationsTypeInv
+    /\ RemovedFromConfigurationTypeInv
 
 \* A set representing requests and responses sent from one server
 \* to another. With CCF, we have message integrity and can ensure unique messages.
 \* Messages only records messages that are currently in-flight, actions should
 \* removed messages once received.
 VARIABLE messages
 
+\* Helper function for checking the type safety of log entries
+EntryTypeOK(entry) ==
+    /\ entry.term \in Nat \ {0}
+    /\ \/ /\ entry.contentType = TypeEntry
+          /\ entry.value \in Nat \ {0}
+       \/ /\ entry.contentType = TypeSignature
+          /\ entry.value = Nil
+       \/ /\ entry.contentType = TypeReconfiguration
+          /\ entry.value \subseteq Servers
+
+AppendEntriesRequestTypeOK(m) ==
+    /\ m.mtype = AppendEntriesRequest
+    /\ m.mprevLogIndex \in Nat
+    /\ m.mprevLogTerm \in Nat
+    /\ m.mcommitIndex \in Nat
+    /\ \A k \in DOMAIN m.mentries: EntryTypeOK(m.mentries[k])
+
+AppendEntriesResponseTypeOK(m) ==
+    /\ m.mtype = AppendEntriesResponse
+    /\ m.msuccess \in BOOLEAN
+    /\ m.mmatchIndex \in Nat
+
+RequestVoteRequestTypeOK(m) ==
+    /\ m.mtype = RequestVoteRequest
+    /\ m.mlastLogTerm \in Nat
+    /\ m.mlastLogIndex \in Nat
+
+RequestVoteResponseTypeOK(m) ==
+    /\ m.mtype = RequestVoteResponse
+    /\ m.mvoteGranted \in BOOLEAN
+
+NotifyCommitMessageTypeOK(m) ==
```

Comment:
FWIW there isn't a distinct message type for this in the implementation, this happens through AppendEntries.leader_commit_idx

Reply:
Yes, thanks for the reminder. I think this will be a fairly big change so out-of-scope for this PR